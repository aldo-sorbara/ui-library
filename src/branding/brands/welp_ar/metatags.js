export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
