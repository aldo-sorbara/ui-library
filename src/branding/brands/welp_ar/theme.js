import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
