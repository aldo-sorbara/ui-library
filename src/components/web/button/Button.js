import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * @render react
 * @name Button
 * @description Button Component for Wenance.
 * @example
 * <ButtonContainer>
 *   <Button data="button"> Button</Button>
 * </ButtonContainer>
 */

export const Button = props => {
  const { children, data, href } = props;
  const Component = href ? StyledLink : StyledButton;
  return (
    <Component {...props} data-w={data}>
      {children}
    </Component>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  data: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  isBlocked: PropTypes.bool,
  modifiers: PropTypes.arrayOf(PropTypes.string)
};

Button.defaultProps = {
  disabled: false,
  isBlocked: false,
  modifiers: []
};

const CommonButton = props => `
  border-radius: ${props.theme.buttons.borderRadius};
  display: ${props.isBlocked ? 'block' : 'inline-block'};
  cursor: pointer;
  font-family: inherit;
  opacity: ${props.disabled ? '0.3' : '1'};
  padding: ${props.theme.spacer.medium} ${props.theme.spacer.wrapperInner};
  text-align: center;
  transition: background-color 0.4s ease-in-out;
  touch-action: manipulation;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: ${props.isBlocked ? '100%' : 'auto'};
  cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
  background-color: ${props.theme.palette.primary.default};
  border: 1px solid transparent;
  color: ${props.theme.palette.common.text.inverse};

  &:hover:enabled {
    background-color: ${props.theme.palette.primary.dark};
  }

  &:focus {
    outline: none;
  }

  &:disabled,
  button[disabled] {
    opacity: 0.3;
  }
`;

CommonButton.defaultProps = {
  theme: {
    palette: {
      common: {
        text: {
          inverse: '#ffffff'
        }
      },
      primary: {
        default: '#0082ff',
        dark: '#3a66ab'
      }
    },
    typography: {
      button: {
        fontSize: '.875rem',
        fontWeight: '700',
        lineHeight: '1.145'
      }
    },
    spacer: {
      medium: '1rem',
      large: '1.5rem',
      button: {
        smallSpacer: '.75rem'
      }
    },
    buttons: {
      borderRadius: '0'
    }
  }
};

const StyledButton = styled.button`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
`;

const StyledLinkButton = props => `
  background: transparent;
  border: 1px solid transparent;
  color: ${props.theme.palette.primary.default};
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }`;

const StyledLink = styled.a`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
  text-decoration: none;
`;

const StyledSmallButton = props => `
  padding: ${props.theme.spacer.button.smallSpacer} ${props.theme.spacer.large};
`;

const StyledInverseButton = props => `
  background: transparent;
  color: ${props.theme.palette.common.text.inverse};
  border: 1px solid ${props.theme.palette.common.text.inverse};
  
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }
`;

const StyledRoundedButton = () => `
  border-radius: 50%;
  width: 50px;
  height: 50px
  padding: 0;

  svg {
    display: block;
    margin: auto;
  }
`;

const modifiers = {
  small: StyledSmallButton,
  rounded: StyledRoundedButton,
  link: StyledLinkButton,
  inverse: StyledInverseButton
};

// ButtonContainer Component
export const ButtonContainer = props => {
  return (
    <StyledButtonContainer {...props}>
      <div>{props.children}</div>
    </StyledButtonContainer>
  );
};

const StyledButtonContainer = styled.div`
  padding: ${props => props.theme.buttons.padding};
`;

StyledButtonContainer.defaultProps = {
  theme: {
    buttons: {
      padding: '0 2rem 2rem'
    }
  }
};
