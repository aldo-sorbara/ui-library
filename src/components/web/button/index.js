export { Button, ButtonContainer } from './Button';
