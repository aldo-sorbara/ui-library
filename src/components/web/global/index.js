export {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from './Global';
