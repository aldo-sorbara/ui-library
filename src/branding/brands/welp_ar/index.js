import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
