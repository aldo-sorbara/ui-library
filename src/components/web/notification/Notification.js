import React from 'react';
import PropTypes from 'prop-types';

import {
  Container,
  Wrapper,
  Headline,
  Description
} from '@bit/wenance.common-ui.web.global';
import { Grid } from '@bit/wenance.common-ui.web.grids';
import { List, ListItem } from '@bit/wenance.common-ui.web.list';
import { ButtonContainer } from '@bit/wenance.common-ui.web.button';
import styled from 'styled-components';

const defaultIcon = () => (
  <IconContainer>
    <svg viewBox='0 0 50 50' style={{ width: '100px', height: '100px' }}>
      <circle cx='25' cy='25' r='25' fill='#7ed321' />
      <path
        fill='none'
        stroke='#fff'
        strokeWidth='2'
        strokeLinecap='round'
        strokeLinejoin='round'
        strokeMiterlimit='10'
        d='M38 15L22 33l-10-8'
      />
    </svg>
  </IconContainer>
);

const selectAsset = (imageUrl, icon) => {
  if (imageUrl) {
    return <img src={imageUrl} style={{ minHeight: 202, marginBottom: 16 }} />;
  } else if (icon) {
    return <IconContainer>{icon}</IconContainer>;
  }
  return defaultIcon();
};
const Notification = props => {
  const {
    header,
    subHeader,
    content,
    imageUrl,
    icon,
    action,
    data,
    padding
  } = props;
  return (
    <NotificationContainer {...props}>
      <Container>
        <Wrapper padding={padding} textAlign='center'>
          {selectAsset(imageUrl, icon)}
          <HeadlineContainer>
            <Headline data-w='notification_title'>{header}</Headline>
          </HeadlineContainer>
          <DescriptionContainer>
            <Description
              style={{ fontSize: '1rem' }}
              data-w='notification_subtitle'
            >
              {subHeader}
            </Description>
            {content && (
              <Description data='notification_content'>{content}</Description>
            )}
          </DescriptionContainer>

          {data && (
            <Grid lg={100} md={100} sm={100} xs={100} data='notification'>
              <List
                data='notification_list'
                style={{ textAlign: 'left', fontSize: '0.75rem' }}
              >
                {data.map(({ text, link }, key) => (
                  <ListItem key={key}>
                    {text}&nbsp;
                    {link && (
                      <a target='_blank' href={link.url}>
                        {link.text}
                      </a>
                    )}
                  </ListItem>
                ))}
              </List>
            </Grid>
          )}
        </Wrapper>
      </Container>
      {action && <ActionContainer>{action}</ActionContainer>}
    </NotificationContainer>
  );
};

Notification.propTypes = {
  header: PropTypes.string.isRequired,
  subHeader: PropTypes.string.isRequired,
  content: PropTypes.string,
  imageUrl: PropTypes.string,
  icon: PropTypes.node,
  action: PropTypes.node,
  data: PropTypes.array
};

Notification.defaultProps = {
  imageUrl: null,
  icon: null,
  content: null,
  data: null,
  action: null
};

const ActionContainer = styled(ButtonContainer)`
  /* stylelint-disable no-empty-block */
`;

const NotificationContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 100vh;
  justify-content: center;
  ${props =>
    props.imageUrl && {
      backgroundImage: 'linear-gradient(to bottom, #ffffff 70%, #ffedd4 100%)'
    }}
  padding: 0 3rem 4.5rem 3rem;
`;

const HeadlineContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-bottom: 1rem;
  margin-top: 1rem;
`;

const DescriptionContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 0 1.75rem 0 1.75rem;
`;

const IconContainer = styled.div`
  align-items: center;
  display: flex;
  flex-direction: column;
  font-size: 4.75rem;
  justify-content: center;
`;

export default Notification;
