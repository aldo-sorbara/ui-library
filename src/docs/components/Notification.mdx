---
name: Notification
menu: Components
route: /notification
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import Notification from '../../components/web/notification';
import { Button } from '../../components/web/button';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';

# Notification

<PropsTable of={Notification} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <Notification
        header={
          welp_ar.wording.onboarding.notifications.productNotAvailable.headline
        }
        subHeader={
          welp_ar.wording.onboarding.notifications.productNotAvailable
            .description
        }
        data={welp_ar.wording.onboarding.notifications.productNotAvailable.list}
        icon={
          welp_ar.wording.onboarding.notifications.productNotAvailable.emoji
        }
        action={
          <Button isBlocked data='notification_button'>
            Siguiente
          </Button>
        }
      />
      <Notification
        header={welp_ar.wording.onboarding.success.headline}
        subHeader={welp_ar.wording.onboarding.success.description}
        data={welp_ar.wording.onboarding.success.list}
        imageUrl={'./public/images/posta-success.png'}
      />
    </>
  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/notification--with-default-props)
