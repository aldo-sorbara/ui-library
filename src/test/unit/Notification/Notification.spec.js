import React from 'react';
import Notification from '../../../components/web/notification';
import { Button } from '../../../components/web/button';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Notification component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Notification
          header={
            welp_ar.wording.onboarding.notifications.productNotAvailable
              .headline
          }
          subHeader={
            welp_ar.wording.onboarding.notifications.productNotAvailable
              .description
          }
          data={
            welp_ar.wording.onboarding.notifications.productNotAvailable.list
          }
        />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a Notification default component', () => {
    const wrapper = mountWithTheme(
      <Notification
        header={
          welp_ar.wording.onboarding.notifications.productNotAvailable.headline
        }
        subHeader={
          welp_ar.wording.onboarding.notifications.productNotAvailable
            .description
        }
      />,
      welp_ar.theme
    );
    expect(wrapper.props().data).toBe(null);
    expect(wrapper.props().imageUrl).toBe(null);
    expect(wrapper.props().icon).toBe(null);
  });
  it('should render a Notification component with icon', () => {
    const wrapper = mountWithTheme(
      <Notification
        header={
          welp_ar.wording.onboarding.notifications.productNotAvailable.headline
        }
        subHeader={
          welp_ar.wording.onboarding.notifications.productNotAvailable
            .description
        }
        data={welp_ar.wording.onboarding.notifications.productNotAvailable.list}
        icon={
          welp_ar.wording.onboarding.notifications.productNotAvailable.emoji
        }
      />,
      welp_ar.theme
    );
    expect(wrapper.props().icon).not.toBe(null);
  });
  it('should render a Notification component with image', () => {
    const wrapper = mountWithTheme(
      <Notification
        header={
          welp_ar.wording.onboarding.notifications.productNotAvailable.headline
        }
        subHeader={
          welp_ar.wording.onboarding.notifications.productNotAvailable
            .description
        }
        data={welp_ar.wording.onboarding.notifications.productNotAvailable.list}
        imageUrl={'./images/posta-success'}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().image).not.toBe(null);
  });
  it('should render a Notification component with a content', () => {
    const wrapper = mountWithTheme(
      <Notification
        header={
          welp_ar.wording.onboarding.notifications.productNotAvailable.headline
        }
        subHeader={
          welp_ar.wording.onboarding.notifications.productNotAvailable
            .description
        }
        data={welp_ar.wording.onboarding.notifications.productNotAvailable.list}
        content={
          welp_ar.wording.onboarding.notifications.productNotAvailable
            .description
        }
        imageUrl={'./images/posta-success'}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().content).not.toBe(null);
  });
  it('should render a Notification component with an action', () => {
    const wrapper = mountWithTheme(
      <Notification
        header={
          welp_ar.wording.onboarding.notifications.productNotAvailable.headline
        }
        subHeader={
          welp_ar.wording.onboarding.notifications.productNotAvailable
            .description
        }
        data={welp_ar.wording.onboarding.notifications.productNotAvailable.list}
        content={
          welp_ar.wording.onboarding.notifications.productNotAvailable
            .description
        }
        imageUrl={'./images/posta-success'}
        action={
          <Button isBlocked data='notification_button'>
            Siguiente
          </Button>
        }
      />,
      welp_ar.theme
    );
    expect(wrapper.props().action).not.toBe(null);
  });
});
