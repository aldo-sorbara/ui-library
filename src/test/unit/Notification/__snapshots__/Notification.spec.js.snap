// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification component should render snapshot 1`] = `
.c1 {
  background-color: transparent;
  margin: 0 auto;
  max-width: unset;
}

.c2 {
  overflow-x: hidden;
  padding: 2rem;
  text-align: center;
}

.c5 {
  margin-bottom: .5rem;
  margin-left: 0;
  margin-top: 0;
  font-size: 1.625rem;
  font-weight: 700;
  line-height: 1.14;
}

.c7 {
  margin-bottom: 1rem;
  margin-top: 0;
  font-size: .875rem;
  font-weight: 400;
  line-height: 1.25;
}

.c8 {
  -webkit-align-items: unset;
  -webkit-box-align: unset;
  -ms-flex-align: unset;
  align-items: unset;
  box-sizing: border-box;
  display: inline-block;
  -webkit-box-pack: unset;
  -webkit-justify-content: unset;
  -ms-flex-pack: unset;
  justify-content: unset;
  -webkit-order: unset;
  -ms-flex-order: unset;
  order: unset;
  padding: 0;
  vertical-align: top;
  width: 100%;
}

.c9 {
  margin: 0;
}

.c10 {
  margin-bottom: .5rem;
  font-size: .875rem;
  font-weight: 400;
  line-height: 1.25;
  padding: 0 0 0 1rem;
}

.c10:last-child {
  margin-bottom: 0;
}

.c10:before {
  top: .25rem;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 3rem 4.5rem 3rem;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-bottom: 1rem;
  margin-top: 1rem;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 1.75rem 0 1.75rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-size: 4.75rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:992px) {
  .c8 {
    width: 100%;
  }
}

@media (max-width:768px) {
  .c8 {
    width: 100%;
  }
}

@media (max-width:576px) {
  .c8 {
    width: 100%;
  }
}

<div
  action={null}
  className="c0"
  content={null}
  data={
    Array [
      Object {
        "text": "Revisá que cuentes con los requisitos",
      },
      Object {
        "link": Object {
          "text": "FAQ",
          "url": "https://www.welp.com/preguntas-frecuentes.html",
        },
        "text": "Si tenés dudas ingresá a ",
      },
      Object {
        "link": Object {
          "text": "Llamanos",
          "url": "tel:08001129357",
        },
        "text": "Si tenés un problema",
      },
    ]
  }
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        <svg
          style={
            Object {
              "height": "100px",
              "width": "100px",
            }
          }
          viewBox="0 0 50 50"
        >
          <circle
            cx="25"
            cy="25"
            fill="#7ed321"
            r="25"
          />
          <path
            d="M38 15L22 33l-10-8"
            fill="none"
            stroke="#fff"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeMiterlimit="10"
            strokeWidth="2"
          />
        </svg>
      </div>
      <div
        className="c4"
      >
        <h1
          className="c5"
          data-w="notification_title"
        >
          ¡Ups!
        </h1>
      </div>
      <div
        className="c6"
      >
        <p
          className="c7"
          data-w="notification_subtitle"
          style={
            Object {
              "fontSize": "1rem",
            }
          }
        >
          No tenemos ningún producto para ofrecerte
        </p>
      </div>
      <div
        className="c8"
        data="notification"
      >
        <ul
          className="c9"
          data="notification_list"
          style={
            Object {
              "fontSize": "0.75rem",
              "textAlign": "left",
            }
          }
        >
          <li
            className="c10"
          >
            Revisá que cuentes con los requisitos
             
          </li>
          <li
            className="c10"
          >
            Si tenés dudas ingresá a 
             
            <a
              href="https://www.welp.com/preguntas-frecuentes.html"
              target="_blank"
            >
              FAQ
            </a>
          </li>
          <li
            className="c10"
          >
            Si tenés un problema
             
            <a
              href="tel:08001129357"
              target="_blank"
            >
              Llamanos
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
